### DISCUSSION TITLE
Support for multidocument YAML output

### DISCUSSION THREAD

**Discussion by mrzor** [cue-lang/cue#2353] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
When using Cue to generate several Kubernetes manifests, I presently have to script around Cue and invoke it _many times_ to output several documents. Invocations looks like `cue export ... -e "_jobs[$i]"` with `$i` being a shell variable.

**Describe the solution you'd like**
I would like the YAML encoder to support outputting multiple documents, but I cannot imagine a clean way for this to happen.

**Describe alternatives you've considered**
I'm thinking of reducing the pain caused by the slowness by outputing only one big YAML document (containing `_jobs` ) and then using `yq` or similar to split it.

**Comment by mxey** [comment, marked as answer]
You can use https://pkg.go.dev/cuelang.org/go@v0.5.0/pkg/encoding/yaml#MarshalStream. See https://github.com/cue-lang/cue/blob/f681271a38ec/doc/tutorial/kubernetes/README.md#dumping-a-yaml-stream for an example.

**Comment by mrzor** [comment]
Thanks a lot mxey I appear to have overlooked this part of the tutorial. This will address my use case just fine. Closing this ticket.
